src/python/example_tensor.py

```diff
@@ -0,0 +1,325 @@
+import sys, os
+import numpy as np
+import singa
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../'))
+from core_pb2 import *
+
+def tuple_to_vector(tshape):
+  if type(tshape) == int:
+    tshape = (tshape, 1)
+  vs = singa.Shape(len(tshape))
+  for i in range(len(tshape)):
+    vs[i] = tshape[i]
+  return vs
+
+''' class and member functions for singa::Tensor
+'''
+class Tensor(object):
+
+  def __init__(self, shape=None, device=None, dtype=kFloat32):
+
+    self.tuple_shape = shape
+    self.device = device
+    self.dtype = dtype
+
+    if shape == None:
+      self.singa_tensor = singa.Tensor()
+    else:
+      vs = tuple_to_vector(shape)
+      if device == None:
+        self.singa_tensor = singa.Tensor(vs, dtype)
+      else:
+        self.singa_tensor = singa.Tensor(vs, device, dtype)
+
+  def copy_Tensor(self, t):
+    self.tuple_shape = t.tuple_shape
+    self.device = t.device
+    self.dtype = t.dtype
+    self.singa_tensor = singa.Tensor(t.singa_tensor)
+
+  def data(self):
+    idx = self.singa_tensor.data_type()
+    if idx == kFloat32:
+      data_array = singa.floatArray_frompointer(self.singa_tensor.floatData())
+      dt = np.float32
+    elif idx == kFloat16:
+      print 'not implemented yet'
+      return
+      #data_array = singa.floatArray_frompointer(self.singa_tensor.floatData())
+      #dt = np.float16
+    elif idx == kInt:
+      data_array = singa.intArray_frompointer(self.singa_tensor.intData())
+      dt = np.int32
+    elif idx == kChar:
+      data_array = singa.charArray_frompointer(self.singa_tensor.charData())
+      dt = np.int8
+    elif idx == kDouble:
+      data_array = singa.doubleArray_frompointer(self.singa_tensor.doubleData())
+      dt = np.float64
+
+    data = [data_array[i] for i in range(self.singa_tensor.Size())]
```

Comment:
what is the type of data_array?
it may be very slow to convert Tensor data into numpy array in this way.